src/components/Sidebar/contentTree.js

```diff
@@ -135,16 +137,34 @@ const ContentTree = ({ edges, location, menuOpen }) => {
   }
 
   const handleLinkClick = () => {
-    const element = document.getElementById('scroll-to-top-main-content');
-    element?.scrollTo(0,0)
+    // Debounce click nav state to prevent unnecessarily moving to active nav button
+    setNavClickStatus(true);
+    setTimeout(() => setNavClickStatus(false), 200);
   };
 
-  useEffect(()=> {
+  const goToActiveNav = () => {
     const sidebarElements = document.getElementsByClassName('ms-Nav-compositeLink is-selected');
     if(sidebarElements.length > 0) {
       sidebarElements[0].scrollIntoView({block: "center", inline: "nearest"});
     }
+  }
+
+  useEffect(()=> {
+    goToActiveNav();
   },[menuOpen]);
+  
+  useEffect(()=> {
+    // Prevent unnecessarily moving to active nav button
+    if(!navClickStatus) {
+      // This active should be executed only other event route path changed, such as clicking item on search result.
+      // Make sure every time that the route path is changed, then go to the active Nav button.
+      setTreeData(calculateNavigationFluentUI(memoizedCalculatedNavigation, location.pathname));
+      setTimeout(() => goToActiveNav(), 200);
```

Comment:
Can we solve this problem with out specific timeout? How do we make sure 200 is good number?
This like a hack for me. Do we have any callback function or other solution without timeout.

Thanks.

Reply:
Yes, it's the hack way. I've fixed it.